pkg/nfd-master/nfd-master.go

```diff
@@ -1219,3 +1218,19 @@ func (m *nfdMaster) instanceAnnotation(name string) string {
 	}
 	return m.args.Instance + "." + name
 }
+
+func (m *nfdMaster) startNfdControllerApi() error {
```

Comment:
Could you split this PR into 2, once for just fixing the resync, and a new once with your proposed enhancements, that way the PR/commit show the actual fix and no more.

Reply:
re-starting the nfd controller API is part of the fix of the resync, and I think extracting the `startNfdControllerApi` in a different method, would remove the redundancy. wdyt?